skimage/measure/_regionprops.py

```diff
@@ -166,6 +228,45 @@ def __init__(self, slice, label, label_image, intensity_image,
         self._cache = {}
         self._ndim = label_image.ndim
 
+        self._extra_properties = {}
+        if extra_properties is None:
+            extra_properties = []
+        for func in extra_properties:
+            name = func.__name__
+            if hasattr(self, name):
+                msg = (
+                    f"Extra property '{name}' is shadowed by existing "
+                    "property and will be inaccessible. Consider renaming it."
+                )
+                warn(msg)
+        self._extra_properties = {
+            func.__name__: func for func in extra_properties
+        }
+
+    def __getattr__(self, attr):
+        if attr in self._extra_properties:
+            func = self._extra_properties[attr]
+            n_args = _infer_number_of_required_args(func)
+            # determine whether func requires intensity image
+            if n_args == 2:
+                if self._intensity_image is not None:
+                    return func(self.image, self.intensity_image)
+                else:
+                    raise AttributeError(
+                        f"intensity image required to calculate {attr}"
+                    )
+            elif n_args == 1:
+                return func(self.image)
+            else:
+                raise AttributeError(f"Function provided for custom regionprop"
+                                     " {attr} takes incorrect number of"
+                                     " arguments."
+                                     )
```

Comment:
```suggestion
                raise AttributeError(
                    f"Function provided for custom regionprop {attr} takes {n_args} "
                    "arguments, but only 1 or 2 are supported."
                )
```

Reply:
(The f-ness of a string does not propagate across multiple lines, fyi.)

Reply:
I did not know that ! Interesting.